[GLib] Fix build without ContentExtensions

blino opened the pull request:
#### 8484c9ba1afef49e25b0ce5c57694004be423c2b
<pre>
[GLib] Fix build without ContentExtensions
<a href="https://bugs.webkit.org/show_bug.cgi?id=240719">https://bugs.webkit.org/show_bug.cgi?id=240719</a>

Reviewed by Adrian Perez de Castro.

Build without ContentExtensions is broken since r241790 because of
API::ContentRuleList usage in WebKitUserContent* API files.

* Source/WebKit/UIProcess/API/glib/WebKitUserContent.cpp:
(_WebKitUserContentFilter::_WebKitUserContentFilter): Make identifier
conditional and reorder fields to make build checks easier to read
* Source/WebKit/UIProcess/API/glib/WebKitUserContentFilterStore.cpp:
(webkitUserContentFilterStoreConstructed):
(webkit_user_content_filter_store_save):
(webkit_user_content_filter_store_save_from_file):
(webkit_user_content_filter_store_remove):
(webkit_user_content_filter_store_load):
(webkit_user_content_filter_store_fetch_identifiers):
* Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp:
(webkit_user_content_manager_add_filter):
(webkit_user_content_manager_remove_filter):
(webkit_user_content_manager_remove_filter_by_id):
(webkit_user_content_manager_remove_all_filters):

Canonical link: <a href="https://commits.webkit.org/251866@main">https://commits.webkit.org/251866@main</a>
</pre>

aperezdc reviewed (commented):
@blino Would you care to elaborate why this is needed? The feature is considered intrinsic to the WPE and GTK ports, and we never intended to support building them without `CONTENT_EXTENSIONS`. It would be extremely awkward to have a part of the public API that may always fail depending on how WebKit was built.

mcatanzaro commented on Source/WebKit/UIProcess/API/glib/WebKitUserContent.cpp:
```diff
@@ -336,8 +336,11 @@ API::UserScript& webkitUserScriptGetUserScript(WebKitUserScript* userScript)
-        , contentRuleList(WTFMove(contentRuleList))
+        :
+#if ENABLE(CONTENT_EXTENSIONS)
+        identifier(contentRuleList->name().utf8()),
```
Indentation is wrong here

aperezdc reviewed (approved):
Thanks for commenting @mcatanzaro! Let's then go with the patch as-is.

@blino Please do fix the issues reported by the style checker before landing, though!

blino committed:
[GLib] Fix build without ContentExtensions
https://bugs.webkit.org/show_bug.cgi?id=240719

Reviewed by Adrian Perez de Castro.

Build without ContentExtensions is broken since r241790 because of
API::ContentRuleList usage in WebKitUserContent* API files.

* Source/WebKit/UIProcess/API/glib/WebKitUserContent.cpp:
(_WebKitUserContentFilter::_WebKitUserContentFilter): Make identifier
conditional and reorder fields to make build checks easier to read
* Source/WebKit/UIProcess/API/glib/WebKitUserContentFilterStore.cpp:
(webkitUserContentFilterStoreConstructed):
(webkit_user_content_filter_store_save):
(webkit_user_content_filter_store_save_from_file):
(webkit_user_content_filter_store_remove):
(webkit_user_content_filter_store_load):
(webkit_user_content_filter_store_fetch_identifiers):
* Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp:
(webkit_user_content_manager_add_filter):
(webkit_user_content_manager_remove_filter):
(webkit_user_content_manager_remove_filter_by_id):
(webkit_user_content_manager_remove_all_filters):

Canonical link: https://commits.webkit.org/251866@main
